src/snowflake/snowpark/ml/transformer.py

```diff
@@ -0,0 +1,499 @@
+            if self._input_cols
+            else None
+        )
+
+    def fit(self, df: DataFrame) -> "MinMaxScaler":
+        super().fit(df)
+
+        scaler_fit(self, df, ["min", "max"])
+        return self
+
+    def transform(self, df: DataFrame) -> DataFrame:
+        """
+        The transformation is given by:
+        X_std = (X - X.min(axis=0)) / (X.max(axis=0) - X.min(axis=0))
+        X_scaled = X_std * (max - min) + min
+        where min, max are decided by feature_range
+        """
+        super().transform(df)
+        states_table = df._session.table(self._states_table_name)
+
+        res_column = df.columns
+
+        for input_col, states_col, output_col in zip(
+            self.input_cols, self._states_table_cols, self.output_cols
+        ):
+            res_column.append(
+                (
+                    (df[input_col] - col(states_col)["min"])
+                    / iff(
+                        (col(states_col)["max"] - col(states_col)["min"]) == 0,
+                        1,
+                        col(states_col)["max"] - col(states_col)["min"],
+                    )
+                    * (self.maximum - self.minimum)
+                    + self.minimum
+                ).as_(output_col)
+            )
+
+        df_transform = states_table.join(df).select(res_column)
+        return df_transform
+
+
+class OneHotEncoder(Transformer):
+    def __init__(
+        self,
+        input_cols: Optional[List[str]] = None,
+        output_cols: Optional[List[str]] = None,
+        category: Union[str, List[List[str]]] = "auto",
+    ) -> None:
+        super().__init__(input_cols, output_cols)
+        self._category = category
+        self._merge = False
+        self._states_table_cols = (
+            [f"states_{input_col}" for input_col in self._input_cols]
+            if self._input_cols
+            else None
+        )
+        self._encoder_table_cols = (
+            [f"states_{input_col}_encoder" for input_col in self._input_cols]
+            if self._input_cols
+            else None
+        )
+        self._category_table_name = {}
+        self._encoder_count_table = random_name_for_temp_object(TempObjectType.TABLE)
+
+    @Transformer.input_cols.setter
+    def input_cols(self, value: List[str]) -> None:
+        self._input_cols = value
+        self._states_table_cols = (
+            [f"states_{input_col}" for input_col in self._input_cols]
+            if self._input_cols
+            else None
+        )
+        self._encoder_table_cols = (
+            [f"states_{input_col}_encoder" for input_col in self._input_cols]
+            if self._input_cols
+            else None
+        )
+
+    def fit(self, df: DataFrame) -> "OneHotEncoder":
+        super().fit(df)
+
+        encoder_fit(self, df, "onehot")
+        return self
+
+    def transform(
+        self,
+        df: DataFrame,
+        sparse: bool = True,
+    ) -> DataFrame:
+        if not self.input_cols:
```

Comment:
You should also call super().transform() and remove the duplicate code, right? The same for ordinal encoder.

Reply:
No, I have to customize it here. If length of output_col is 1 and not equals to input_col, super().transform() will raise a ValueError directly.

Reply:
I see, then super().transform() shouldn't check this. The method in the base class should work for all subclasses.